From: Irfan Mohammad <irfan@graphiant.com>
Date: Thu, 17 Nov 2022 09:17:24 +0000
Subject: Enable tests in package build

---
 CMakeLists.txt       |  2 +-
 tests/CMakeLists.txt | 10 ++++++++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f0d952d..8e8e7af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,7 +90,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wpedantic -std=c11")
 #
 # options
 #
-if(("${BUILD_TYPE_UPPER}" STREQUAL "DEBUG") OR ("${BUILD_TYPE_UPPER}" STREQUAL "RELWITHDEBINFO"))
+if(("${BUILD_TYPE_UPPER}" STREQUAL "DEBUG") OR ("${BUILD_TYPE_UPPER}" STREQUAL "RELWITHDEBINFO") OR ("${BUILD_TYPE_UPPER}" STREQUAL "PACKAGE"))
     option(ENABLE_TESTS "Build tests" ON)
     option(ENABLE_VALGRIND_TESTS "Build tests with valgrind" ON)
 else()
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 9a026e0..512fcfa 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -68,12 +68,15 @@ if(ENABLE_TESTS)
             "CMOCKA_TEST_ABORT=1"
             "SYSREPO_REPOSITORY_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}"
             "SYSREPO_SHM_PREFIX=_tests_sr_${test_name}"
+            "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}"
         )
 
         # avoid loading installed plugins by sysrepo-plugind
         if(${test_name} STREQUAL "test_rotation")
             set_property(TEST ${test_name} APPEND PROPERTY ENVIRONMENT
-                "SRPD_PLUGINS_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}/testing")
+                "SRPD_PLUGINS_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}/testing"
+                "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}"
+                )
         endif()
 
         if(${CMAKE_VERSION} VERSION_GREATER "3.7")
@@ -99,12 +102,15 @@ if(ENABLE_VALGRIND_TESTS)
         set_property(TEST ${test_name} APPEND PROPERTY ENVIRONMENT
             "SYSREPO_REPOSITORY_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}"
             "SYSREPO_SHM_PREFIX=_tests_sr_${test_name}"
+            "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}"
         )
 
         # avoid loading installed plugins by sysrepo-plugind
         if(${test_name} STREQUAL "test_rotation_valgrind")
             set_property(TEST ${test_name} APPEND PROPERTY ENVIRONMENT
-                "SRPD_PLUGINS_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}/testing")
+                "SRPD_PLUGINS_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}/testing"
+                "LD_LIBRARY_PATH=${CMAKE_BINARY_DIR}"
+            )
         endif()
 
         if(${CMAKE_VERSION} VERSION_GREATER "3.7")
