From: Irfan Mohammad <irfan@graphiant.com>
Date: Wed, 30 Nov 2022 10:36:59 +0000
Subject: Accept empty xpath "" as NULL

---
 src/sysrepo.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/sysrepo.c b/src/sysrepo.c
index 947481c..c8c8780 100644
--- a/src/sysrepo.c
+++ b/src/sysrepo.c
@@ -5854,6 +5854,11 @@ sr_module_change_subscribe(sr_session_ctx_t *session, const char *module_name, c
     SR_CHECK_ARG_APIRET(!session || !SR_IS_STANDARD_DS(session->ds) || SR_IS_EVENT_SESS(session) || !module_name ||
             !callback || !subscription, session, err_info);
 
+    /* xpath = "" is invalid, implicitly convert it to NULL to avoid errors */
+    if (xpath && !xpath[0]) {
+        xpath = NULL;
+    }
+
     SR_MODINFO_INIT(mod_info, session->conn, session->ds, session->ds == SR_DS_OPERATIONAL ? SR_DS_RUNNING : session->ds, 0);
 
     conn = session->conn;
@@ -6021,6 +6026,11 @@ sr_module_change_sub_modify_xpath(sr_subscription_ctx_t *subscription, uint32_t
 
     SR_CHECK_ARG_APIRET(!subscription || !sub_id, NULL, err_info);
 
+    /* xpath = "" is invalid, implicitly convert it to NULL to avoid errors */
+    if (xpath && !xpath[0]) {
+        xpath = NULL;
+    }
+
     /* SUBS WRITE LOCK */
     if ((err_info = sr_rwlock(&subscription->subs_lock, SR_SUBSCR_LOCK_TIMEOUT, SR_LOCK_WRITE, subscription->conn->cid,
             __func__, NULL, NULL))) {
@@ -6075,7 +6085,7 @@ _sr_get_changes_iter(sr_session_ctx_t *session, const char *xpath, int dup, sr_c
 {
     sr_error_info_t *err_info = NULL;
 
-    SR_CHECK_ARG_APIRET(!session || !SR_IS_EVENT_SESS(session) || !xpath || !iter, session, err_info);
+    SR_CHECK_ARG_APIRET(!session || !SR_IS_EVENT_SESS(session) || !xpath || !xpath[0] || !iter, session, err_info);
 
     if ((session->ev != SR_SUB_EV_ENABLED) && (session->ev != SR_SUB_EV_DONE) && !session->dt[session->ds].diff) {
         sr_errinfo_new(&err_info, SR_ERR_INVAL_ARG, "Session without changes.");
