From: Irfan Mohammad <irfan@graphiant.com>
Date: Sun, 11 Dec 2022 00:28:01 +0000
Subject: sysrepocfg: change default file format to JSON

---
 src/executables/sysrepocfg.c | 61 ++++++++++++++++++++++----------------------
 1 file changed, 30 insertions(+), 31 deletions(-)

diff --git a/src/executables/sysrepocfg.c b/src/executables/sysrepocfg.c
index 508dfe0..6f6d547 100644
--- a/src/executables/sysrepocfg.c
+++ b/src/executables/sysrepocfg.c
@@ -36,6 +36,8 @@
 #include "bin_common.h"
 #include "sysrepo.h"
 
+#define DEFAULT_FORMAT LYD_JSON
+
 sr_log_level_t log_level = SR_LL_ERR;
 
 static void
@@ -83,7 +85,7 @@ help_print(void)
             "  -m, --module <module-name>   Module to be operated on, otherwise it is operated on full datastore.\n"
             "                               Accepted by import, export, edit, copy-from op.\n"
             "  -x, --xpath <xpath>          XPath to select. Accepted by export op.\n"
-            "  -f, --format <format>        Data format to be used, by default based on file extension or \"xml\" if not applicable\n"
+            "  -f, --format <format>        Data format to be used, by default based on file extension or \"json\" if not applicable\n"
             "                               (\"xml\", \"json\", or \"lyb\"). Accepted by import, export, edit, rpc,\n"
             "                               notification, copy-from op.\n"
             "  -l, --lock                   Lock the specified datastore for the whole operation. Accepted by edit op.\n"
@@ -225,6 +227,28 @@ enum data_type {
     DATA_OPER
 };
 
+static LYD_FORMAT
+learn_lyd_format(const char *file_path)
+{
+    char *ptr = NULL;
+    LYD_FORMAT format = DEFAULT_FORMAT;
+    /* learn format */
+    if (!file_path) {
+        return format;
+    }
+
+    ptr = strrchr(file_path, '.');
+    if (ptr && !strcmp(ptr, ".xml")) {
+        format = LYD_XML;
+    } else if (ptr && !strcmp(ptr, ".json")) {
+        format = LYD_JSON;
+    } else if (ptr && !strcmp(ptr, ".lyb")) {
+        format = LYD_LYB;
+    }
+
+    return format;
+}
+
 static int
 step_load_data(const struct ly_ctx *ly_ctx, const char *file_path, LYD_FORMAT format, enum data_type data_type,
         int not_strict, int opaq, struct lyd_node **data)
@@ -234,26 +258,6 @@ step_load_data(const struct ly_ctx *ly_ctx, const char *file_path, LYD_FORMAT fo
     int parse_flags;
     LY_ERR lyrc = 0;
 
-    /* learn format */
-    if (format == LYD_UNKNOWN) {
-        if (!file_path) {
-            error_print(0, "When reading data from STDIN, format must be specified");
-            return EXIT_FAILURE;
-        }
-
-        ptr = strrchr(file_path, '.');
-        if (ptr && !strcmp(ptr, ".xml")) {
-            format = LYD_XML;
-        } else if (ptr && !strcmp(ptr, ".json")) {
-            format = LYD_JSON;
-        } else if (ptr && !strcmp(ptr, ".lyb")) {
-            format = LYD_LYB;
-        } else {
-            error_print(0, "Failed to detect format of \"%s\"", file_path);
-            return EXIT_FAILURE;
-        }
-    }
-
     /* get input */
     if (file_path) {
         lyrc = ly_in_new_filepath(file_path, 0, &in);
@@ -313,8 +317,6 @@ step_create_input_file(LYD_FORMAT format, char *tmp_file)
     if (format == LYD_LYB) {
         error_print(0, "LYB binary format cannot be opened in a text editor");
         return EXIT_FAILURE;
-    } else if (format == LYD_UNKNOWN) {
-        format = LYD_XML;
     }
 
 #ifdef SR_HAVE_MKSTEMPS
@@ -397,10 +399,6 @@ op_export(sr_session_ctx_t *sess, const char *file_path, const char *module_name
     char *str;
     int r;
 
-    if (format == LYD_UNKNOWN) {
-        format = LYD_XML;
-    }
-
     if (file_path) {
         file = fopen(file_path, "w");
         if (!file) {
@@ -570,11 +568,8 @@ op_rpc(sr_session_ctx_t *sess, const char *file_path, const char *editor, LYD_FO
             break;
         }
     }
+
     if (node) {
-        if (format == LYD_UNKNOWN) {
-            /* use XML as the default */
-            format = LYD_XML;
-        }
         lyd_print_file(stdout, output->tree, format, wd_opt);
     }
     sr_release_data(output);
@@ -1092,6 +1087,10 @@ main(int argc, char **argv)
         goto cleanup;
     }
 
+    if (format == LYD_UNKNOWN) {
+        format = learn_lyd_format(file_path);
+    }
+
     /* perform the operation */
     switch (operation) {
     case 'I':
