From: Irfan Mohammad <irfan@graphiant.com>
Date: Sun, 26 Feb 2023 16:44:56 +0000
Subject: ncid compatibility

---
 src/sysrepo.c | 24 ++++++++++++++++++++++++
 src/sysrepo.h |  3 +++
 2 files changed, 27 insertions(+)

diff --git a/src/sysrepo.c b/src/sysrepo.c
index e7dadcf..b5e7985 100644
--- a/src/sysrepo.c
+++ b/src/sysrepo.c
@@ -7314,3 +7314,27 @@ cleanup:
     sr_lycc_unlock(conn, SR_LOCK_READ, 0, __func__);
     return sr_api_ret(session, err_info);
 }
+
+API void
+sr_session_set_nc_id(sr_session_ctx_t *session, uint64_t nc_id)
+{
+    if (!session) {
+        return;
+    }
+    sr_session_push_orig_data(session, sizeof nc_id, &nc_id);
+}
+
+API uint64_t
+sr_session_get_event_nc_id(sr_session_ctx_t *session)
+{
+    uint64_t * nc_id = NULL;
+    uint32_t size = 0;
+    if (!session || !session->ev) {
+        return 0;
+    }
+
+    sr_session_get_orig_data(session, 0, &size, (const void **)&nc_id);
+    
+    return nc_id ? *nc_id : 0;
+}
+
diff --git a/src/sysrepo.h b/src/sysrepo.h
index 8a6a70b..9a5773d 100644
--- a/src/sysrepo.h
+++ b/src/sysrepo.h
@@ -1869,6 +1869,9 @@ void srplg_errinfo_free(sr_error_info_t **err_info);
  */
 void srplg_log(const char *plg_name, sr_log_level_t ll, const char *format, ...) _FORMAT_PRINTF(3, 4);
 
+uint64_t sr_session_get_event_nc_id(sr_session_ctx_t *session);
+void sr_session_set_nc_id(sr_session_ctx_t *session, uint64_t nc_sid);
+
 #ifdef __cplusplus
 }
 #endif
