From: Irfan Mohammad <irfan@graphiant.com>
Date: Fri, 17 Mar 2023 04:22:17 +0000
Subject: Fix ncid originator name

---
 src/sysrepo.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/sysrepo.c b/src/sysrepo.c
index c863ad6..7d4d05d 100644
--- a/src/sysrepo.c
+++ b/src/sysrepo.c
@@ -7899,9 +7899,17 @@ cleanup:
 API void
 sr_session_set_nc_id(sr_session_ctx_t *session, uint64_t nc_id)
 {
+    extern char *__progname;
+
     if (!session) {
         return;
     }
+    if (!session->orig_name) {
+        /* TODO FIXME ENG-4345 */
+       sr_session_set_orig_name(session, __progname);
+    }
+    /* Need to remove all previous data */
+    sr_session_del_orig_data(session);
     sr_session_push_orig_data(session, sizeof nc_id, &nc_id);
 }
 
@@ -7915,7 +7923,6 @@ sr_session_get_event_nc_id(sr_session_ctx_t *session)
     }
 
     sr_session_get_orig_data(session, 0, &size, (const void **)&nc_id);
-    
     return nc_id ? *nc_id : 0;
 }
 
