From: IrfanMohammad <irfan@graphiant.com>
Date: Wed, 26 Jul 2023 12:23:28 +0000
Subject: Add PTHREAD_MUTEX_ERRORCHECK attr

---
 src/common.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/common.c b/src/common.c
index 2fa107a..c34e44b 100644
--- a/src/common.c
+++ b/src/common.c
@@ -2119,6 +2119,12 @@ _sr_mutex_init(pthread_mutex_t *lock, int shared, int robust)
             return err_info;
         }
 
+        if ((ret = pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ERRORCHECK))) {
+            pthread_mutexattr_destroy(&attr);
+            sr_errinfo_new(&err_info, SR_ERR_SYS, "mutex error_check failed (%s).", strerror(ret));
+            return err_info;
+        }
+
         if ((ret = pthread_mutex_init(lock, &attr))) {
             pthread_mutexattr_destroy(&attr);
             sr_errinfo_new(&err_info, SR_ERR_SYS, "Initializing pthread mutex failed (%s).", strerror(ret));
