From: IrfanMohammad <irfan@graphiant.com>
Date: Wed, 26 Jul 2023 12:26:55 +0000
Subject: Log error when subscr thread exits

---
 src/shm_sub.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/shm_sub.c b/src/shm_sub.c
index a6dbad5..c8acfbb 100644
--- a/src/shm_sub.c
+++ b/src/shm_sub.c
@@ -4853,6 +4853,8 @@ sr_shmsub_listen_thread(void *arg)
              * another event is generated, our event pipe will not get notified */
             continue;
         } else if (ret) {
+            sr_errinfo_new(&err_info, ret, "sr_subscription_process_events failed with %d", ret);
+            sr_errinfo_free(&err_info);
             goto error;
         }
 
