From: IrfanMohammad <irfan@graphiant.com>
Date: Wed, 23 Aug 2023 09:01:25 +0000
Subject: cache invalidation without file m_time

---
 src/plugins/ds_json.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugins/ds_json.c b/src/plugins/ds_json.c
index ac2dbac..d18225e 100644
--- a/src/plugins/ds_json.c
+++ b/src/plugins/ds_json.c
@@ -213,9 +213,9 @@ srpds_json_store_(const struct lys_module *mod, sr_datastore_t ds, const struct
         goto cleanup;
     }
 
+cleanup:
     free(str);
 
-cleanup:
     /* delete the backup file */
     if (backup && (unlink(bck_path) == -1)) {
         srplg_log_errinfo(&err_info, srpds_name, NULL, SR_ERR_SYS, "Failed to remove backup \"%s\" (%s).", bck_path,
